src/coreclr/inc/corinfo.h

```diff
@@ -356,6 +356,95 @@ enum StructFloatFieldInfoFlags
     STRUCT_HAS_8BYTES_FIELDS_MASK = (STRUCT_FIRST_FIELD_SIZE_IS8 | STRUCT_SECOND_FIELD_SIZE_IS8),
 };
 
+// Bitfields for FpStructInRegistersInfo::flags
+namespace FpStruct
+{
+    enum class IntKind
+    {
+        Integer,
+        GcRef,
+        GcByRef,
+    };
+
+    enum Flags
+    {
+        // Positions of flags and bitfields
+        PosOnlyOne      = 0,
+        PosBothFloat    = 1,
+        PosFloatInt     = 2,
+        PosIntFloat     = 3,
+        PosSizeShift1st = 4, // 2 bits
+        PosSizeShift2nd = 6, // 2 bits
+        PosIntFieldKind = 8, // 2 bits
+
+        UseIntCallConv = 0, // struct is passed according to integer calling convention
+
+        // The flags and bitfields
+        OnlyOne          =    1 << PosOnlyOne,      // has only one field, which is floating-point
+        BothFloat        =    1 << PosBothFloat,    // has two fields, both are floating-point
+        FloatInt         =    1 << PosFloatInt,     // has two fields, 1st is floating and 2nd is integer
+        IntFloat         =    1 << PosIntFloat,     // has two fields, 2nd is floating and 1st is integer
+        SizeShift1stMask = 0b11 << PosSizeShift1st, // log2(size) of 1st field
+        SizeShift2ndMask = 0b11 << PosSizeShift2nd, // log2(size) of 2nd field
+        IntFieldKindMask = 0b11 << PosIntFieldKind, // the kind of the integer field (FpStruct::IntKind)
+        // Note: flags OnlyOne, BothFloat, FloatInt, and IntFloat are mutually exclusive
+    };
+}
+
+// On RISC-V and LoongArch a struct with up to two non-empty fields, at least one of them floating-point,
+// can be passed in registers according to hardware FP calling convention. FpStructInRegistersInfo represents
+// passing information for such parameters.
+struct FpStructInRegistersInfo
```

Comment:
@LuckyXu-HF @shushanhf I still need to update LoongArch API to return the same struct, putting up PR now so CI can start working.

PTAL if there's anything to improve, I'd like LoongArch to share the same container due to similarities in FP struct passing. Offsets for LoongArch could be smaller ints but most decision-making is done based on flags anyway so I don't think it will be a problem.

Reply:
Yes, the current ABI's implementation doesn't support the extra placement info for LA64/RV64.
Thanks for your amendment for LA64.

Reply:
OK, LoongArch updated. But I don't have a good way to test it so PTAL.

Reply:
OK, thanks.
We will test it on LA64.